ui/src/containers/FullActivitiesList/index.js

```diff
@@ -0,0 +1,208 @@
+import React from 'react';
+import ModelOptionList from 'ui/components/AutoComplete2/Options/ModelOptionList/ModelOptionList';
+// import SingleInput from 'ui/components/AutoComplete2/Inputs/SingleInput/SingleInput';
+import AutoComplete2 from 'ui/components/AutoComplete2';
+import languageResolver from 'ui/utils/languageResolver';
+import { compose, withState, withProps, shouldUpdate } from 'recompose';
+import { Map, List, fromJS } from 'immutable';
+import { connect } from 'react-redux';
+import { addTokenToQuery, getCriteria, changeCriteria } from 'ui/utils/queries';
+import { valueToCriteria } from 'ui/redux/modules/queryBuilder';
+import { modelsSchemaIdSelector } from 'ui/redux/selectors';
+import { createSelector } from 'reselect';
+import { withModels } from 'ui/utils/hocs';
+import shallowEqualObjects from 'shallow-equal/objects';
+import FullActivitiesInput from './FullActivitiesInput';
+
+const fullActivitiesList = ({
+  selectedOption,
+  setSearchString,
+  setSearchFilter,
+  valuesFilter,
+  model,
+  models, // remove currently selected model
+  updateVisualisationModel,
+}) => {
+  const onSelectOption = ({
+    onBlur,
+  }) => (e) => {
+
+    let deselectedFilters = model.get('filters');
+    if (models.size > 0) {
```

Comment:
This if should be unnecessary since it wouldn't map over anything if there were no items to map over, it would just return an empty array/list.

Reply:
Think it is necessary, as it's not models we are mapping over.

Reply:
Ah yeah good point, misinterpreted the condition initially.